app/services/cognito/sign_in_user.rb

```diff
@@ -33,6 +33,9 @@ def call
     rescue Aws::CognitoIdentityProvider::Errors::UserNotFoundException
       @error = I18n.t('base.users.sign_in_error')
       errors.add(:base, @error)
+    rescue Aws::CognitoIdentityProvider::Errors::NotAuthorizedException => e
```

Comment:
My only concern with this is if AWS were to change the error message for whatever reason.That being said, I don't see why they would and I think this is the only way you can really tell if the number of attempts has been exceeded.